### ISSUE TITLE
mseedscan does only a single channel

### ISSUE THREAD

**Issue by FMassin** [EarthScope/ringserver#52]
Hello

I don't mean to declare a real issue, at least not specifically about the code, but more generally about the mseedscan feature. I guess the problem is most probably on my side.
  
I receive data from an external record stream source. I write them in 512b mseed using `slinktool` as follows:
```bash
-rw-r--r--  1 fred  staff   3.0K Jun  8 08:52 data_miniseed//2024/CH/FIESA/HGE.D/CH.FIESA..HGE.D.2024.160
-rw-r--r--  1 fred  staff   3.5K Jun  8 08:52 data_miniseed//2024/CH/FIESA/HHZ.D/CH.FIESA..HHZ.D.2024.160
-rw-r--r--  1 fred  staff   512B Jun  8 08:52 data_miniseed//2024/CH/FIESA/BLN.D/CH.FIESA..BLN.D.2024.160
-rw-r--r--  1 fred  staff   4.0K Jun  8 08:52 data_miniseed//2024/CH/FIESA/HHE.D/CH.FIESA..HHE.D.2024.160
-rw-r--r--  1 fred  staff   512B Jun  8 08:52 data_miniseed//2024/CH/FIESA/BHN.D/CH.FIESA..BHN.D.2024.160
-rw-r--r--  1 fred  staff   3.0K Jun  8 08:52 data_miniseed//2024/CH/FIESA/HGZ.D/CH.FIESA..HGZ.D.2024.160
-rw-r--r--  1 fred  staff   4.0K Jun  8 08:52 data_miniseed//2024/CH/FIESA/HHN.D/CH.FIESA..HHN.D.2024.160
-rw-r--r--  1 fred  staff   512B Jun  8 08:52 data_miniseed//2024/CH/FIESA/BLE.D/CH.FIESA..BLE.D.2024.160
-rw-r--r--  1 fred  staff   512B Jun  8 08:52 data_miniseed//2024/CH/FIESA/BHE.D/CH.FIESA..BHE.D.2024.160
-rw-r--r--  1 fred  staff   3.5K Jun  8 08:52 data_miniseed//2024/CH/FIESA/HGN.D/CH.FIESA..HGN.D.2024.160
-rw-r--r--  1 fred  staff   512B Jun  8 08:52 data_miniseed//2024/CH/FIESA/BHZ.D/CH.FIESA..BHZ.D.2024.160
-rw-r--r--  1 fred  staff   512B Jun  8 08:52 data_miniseed//2024/CH/FIESA/BLZ.D/CH.FIESA..BLZ.D.2024.160
```
I also have the `ringserver` (branch `origin/45-docker-container-does-not-compile`) docker image with the following config:
```bash
diff --git a/doc/ring.conf b/doc/ring.conf
index 474b877..621195e 100644
--- a/doc/ring.conf
+++ b/doc/ring.conf
@@ -264,4 +264,5 @@ ListenPort 18000
 # contain spaces.
 # See the ringserver(1) man page for more details.
 
+MSeedScan /data/miniseed/ StateFile=/tmp/scan.state
 #MSeedScan <directory> [StateFile=scan.state] [Match=pattern] [Reject=pattern] [InitCurrentState=y]
```
running in the following container:
```bash
❯ docker run -dit --name ringserver -v $(pwd)/data_miniseed:/data/miniseed/  -p 64014:18000  ringserver:latest
❯ slinktool -Q -v localhost:64014
slinktool version: 4.1b-sc3
[localhost:64014] network socket opened
[localhost:64014] connected to: SeedLink v3.1 (2020.075 RingServer) 
[localhost:64014] capabilities: SLPROTO:3.1 CAP EXTREPLY NSWILDCARD BATCH WS:13
[localhost:64014] organization: Ring Server
[localhost:64014] requesting INFO level STREAMS
2024.160.08:52:37.0, seq 0, Received Info (terminated) blockette
CH FIESA    HGN D 2024-06-08 06:52:21  -  2024-06-08 06:52:33
[localhost:64014] network socket closed
```
As can be seen in the slinktool output, there is only one channel in seedlink.

Where's my mistake? Is there somewhere an example of how to use the mseedscan feature in ringserver? Or is ther something unexpected happening?

Cheers

Fred

**Comment by chad-earthscope** [comment]
Hi Fred,

Kind of a long shot, but how long did you wait between starting the docker container and running slinktool?  If it was short there may not have been time to scan and load the data yet.

The biggest complicating issue here is using docker at all.  I strongly recommend running the same commands with ringserver directly and seeing if that works before trying with a dockerized version.

**Comment by FMassin** [comment]
Hi @chad-earthscope !

I wait minutes... So you don't see another issue with my usage than docker. Maybe the docker share volume is problematic.

**Comment by chad-earthscope** [comment]
I wonder if this is fixed in v4.0.0.
Some numerical instability in an internal hash function has been addressed, and would lead to weird errors like all source IDs being the same.

**Comment by FMassin** [comment]
Hello @chad-earthscope 

I took a different path to fulfil my purpose for opening this. And I totally lost track. I doubt I can find the time to redo it, sorry.
